### ISSUE TITLE
Retest: CO2 Calculation View is available for locations that don't report CO2C Derived Hourly Data

### ISSUE THREAD

**Issue by esaber76** [US-EPA-CAMD/PI-1#6046]
Emissions: The CO2 Calculation View is an available template for locations where it does not apply. This should only apply to locations with emissions data that report CO2C DHV records (CO2 derived from O2 component).

Example where it should not show up: ORIS 1356, CS023 2023 Q3 emissions
Example where it should still show up after changes: ORIS 58054, ST01 2023 Q3 emissions

Also have seen this showing up in error for other locations that report all CEMS.

**Comment by ibarra-michelle** [comment]
@annalbrecht  we discussed this ticket internally and would like to deprioritize this to Phase 1B. #6046 is appears to be related to #5803 Bug: Viewing Emissions Emissions Data. Should ticket #5803 be managed separately to 5803, together or other?

If the dev team has any questions, we can discuss them at the upcoming stand-up meeting.

FYA - This ticket was deprioritized since it isn't a blocker for industry submission and is a low priority compared with other development work.

**Comment by ibarra-michelle** [comment]
@annalbrecht - could we relabel this from P-1B to 2025 (Q1 or Q2)? Internally, we discussed this ticket and are prioritizing other functionality. If the dev team has any questions, we can discuss them at the next meeting.

Also, should we link this ticket #6046 to #5803 Bug: Viewing Emissions Data? They seem related.

**Comment by ibarra-michelle** [comment]
@annalbrecht Is ticket #6249 [Enhancement: Limit available emissions templates displayed in public view](https://github.com/US-EPA-CAMD/easey-ui/issues/6246#top) and #6046 CO2 Calculation View is available for locations that don't report CO2C Derived Hourly Data related? Should they be tracked together?

@ibarra-michelle  ERG thinks these tickets should be kept separate. Also, the CO2C issue may have been fixed by the refactor emissions views ticket (#5793), but it needs more in-depth testing to confirm.

**Comment by ibarra-michelle** [comment]
> @annalbrecht Is ticket #6249 [Enhancement: Limit available emissions templates displayed in public view](https://github.com/US-EPA-CAMD/easey-ui/issues/6246#top) and #6046 CO2 Calculation View is available for locations that don't report CO2C Derived Hourly Data related? Should they be tracked together?
> 
> @ibarra-michelle ERG thinks these tickets should be kept separate. Also, the CO2C issue may have been fixed by the refactor emissions views ticket (#5793), but it needs more in-depth testing to confirm.

That sounds good, thanks @annalbrecht

**Comment by annalbrecht** [comment]
Retest in P-1B after #5793 is confirmed in Dev. If still an issue, move to 2025.

**Comment by esaber76** [comment]
Blocked by #6410. I was going to test this separately from the remaining work in 6410, but think these should stay together just in case... When 6410 is ready to go to Beta, this can follow.

**Comment by esaber76** [comment]
### Acceptance Tests:

**Public View:**
1. ORIS 1356, CS023, 2, 3: Selecting the CO2 Calculation View for 2023 Q3 emissions data for all 3 locations does not return any results and the proper message is displayed.
2. ORIS 58054, ST01: Selecting the CO2 Calculation View for 2023 Q3 emissions data does return data.

**Workspace View:**
1. ORIS 1356, CS023, 2, 3: Import 2023 Q3 emissions data from historical and confirm the CO2 Calculation View is not an available template for all 3 locations.
2. ORIS 58054, ST01: Import 2023 Q3 emissions data from historical and confirm the CO2 Calculation View does return data.

**Comment by ntknguyen** [comment]
Verified the issue is corrected in workspace, but incorrect in public view as listed in the acceptance criteria.

**Comment by esaber76** [comment]
> Verified the issue is corrected in workspace, but incorrect in public view. <img alt="Image" width="509" src="https://private-user-images.githubusercontent.com/17596531/394938729-1e8d2765-8c03-4d11-8fa6-cf37074475e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5NTYwODEsIm5iZiI6MTczMzk1NTc4MSwicGF0aCI6Ii8xNzU5NjUzMS8zOTQ5Mzg3MjktMWU4ZDI3NjUtOGMwMy00ZDExLThmYTYtY2YzNzA3NDQ3NWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDIyMjMwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyNDkyYmM3MDZkYTNlODlkMDFhNTZiYzAyYjY5MjFlM2ExMTQ5N2RjMTYyZjNlMzY2ZTYxM2FjZWY2YjliMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TKwItG79DLjxi3NOWX1ZUIs_qpGtVT4k7m6RAeV9XWw">

@ntknguyen - the view is available in public view, but if you select it and select the Apply Filter(s) button, no data should be returned with the appropriate message. There is a ticket in the backlog somewhere to limit the public view display to only show views with data.
